### PR TITLE
Add preserve input text inference

### PR DESCRIPTION
**Pull request by Ssukriti**
for: https://github.com/caikit/caikit-nlp/issues/243

adds a flag to preserve or remove input text from output of CAUSAL models (**local inferencing** of prompt tuning and generation)

this will achieve functional parity with TGIS

The change will enable local evaluation of causal models without TGIS

### CODE REVIEW

File: caikit_nlp/toolkit/text_generation/model_run_utils.py [hunk 1]
```diff
@@ -251,14 +266,37 @@ def generate_text_func(
...
         input_token_count=input_token_count,
         seed=seed,
     )
 
 
+def __postprocess_remove_input_text(tokenizer, preds, inputs, task_type):
+    """For Causal LM task types, preserve_input_text set to False will
+    remove the input text from generated output.
+    """
+    if task_type == "CAUSAL_LM":
```
Comment (Ssukriti) [inline caikit_nlp/toolkit/text_generation/model_run_utils.py:281]: HF pipeline has implemented this for Causal LMs generation pipelines https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/text_generation.py#L287 

I took the same logic as it was suggested by other stack overflow posts too. I set the flag to True by default as generate() returns full text by default. 

I have tested this with fine tuning on llama model with different texts and found it to work as expected. Unit tests show the same

**Comment by Ssukriti** [comment]
actually hold on, making some changes

**Comment by Ssukriti** [comment]
@gkumbhat its ready now when you have a chance, thanks

**Review by gkumbhat — APPROVED** [review]
LGTM. Thanks Sukriti

**Pull request merged**